tools/docker/boardfarm-ci/Dockerfile

```diff
@@ -1,41 +1,29 @@
 FROM python:3.8-slim-buster
+# FROM tiangolo/docker-with-compose
 
 RUN apt-get update \
-    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
-    apt-transport-https \
-    ca-certificates curl \
-    curl \
-    gcc \
-    git \
-    gnupg \
-    gnupg-agent \
-    libsnmp-dev \
-    netcat \
-    software-properties-common \
-    wireshark-common \
-    && rm -rf /var/lib/apt/lists/*
+&& apt-get install gcc libsnmp-dev -y \
+&& apt-get clean
 
 COPY requirements.txt /app/requirements.txt
 
 WORKDIR app
 RUN pip3 install -r requirements.txt
 
-# TODO: what needs this?
-#RUN pip3 install jsonschema distro
+RUN apt-get install apt-transport-https ca-certificates curl gnupg-agent software-properties-common -y
```

Comment:
This whole commit is not very nice, because it splits the package installation for no reason, and also runs `apt-get install` without previously running `update`, so it will break if apt's cache is cached for long enough (say if docker tries to rebuild from here in 3 month).

Reply:
There was a line with apt-get update above, though not very obvious

It would make sense to put all apt-get installations on one place, but the ordering on the file is as follows:

- Debian dependencies to install the python packages
- Installation of the python packages (pip3 install...)
- Debian dependencies to install the docker and docker compose binaries
- Installation of docker and docker compose
- Debian dependencies for the tests themselves (wireshark, etc)
- Installation of boardfarm

Given the layered nature of docker this allows for better iterations when adding changes. I do not think putting al the packages for apt-get in one line would yield significant performance benefits.

But the good ordering, from the less likely to change to the most, would be:

- Dependencies to install docker and docker-compose
- Installation of docker and docker-compose 
- Installation of boardfarm (from github)
- Debian dependencies to install python packages (for tests)
- Installation of the python packages (for tests)
- Debian dependencies used by the tests themselves

I have commented the file to make clearer this rationale (dd55701f)

Reply:
You missed my point about Docker's caching.
It's not about performances at all, personally here I don't care where you start rebuilding from, the image doesn't take that much time to build anyway.
Let me take a very concrete example that highlights why the current version is wrong.
- we do run `apt-get` update at the beginning and install some packages, everything is fine.
- we install a specific version of boardfarm and this layer is stored in cache: 
  ```
  RUN git clone https://github.com/mattsm/boardfarm.git \
    && cd boardfarm \
    && git checkout 100521fde1fb67536682cafecc2f91a6e2e8a6f8 \
    && python3 setup.py install
  ```

- next, we install some packages, and this is another layer stored in cache:
  ```
  RUN apt-get install gcc libsnmp-dev -y && apt-get clean
  ```

- now say that next month you want to update the boardfarm version you use, so you change the hash.
- docker is going to re-run the git clone thing, everything is fine.
- then comes the call to `apt-get install gcc ...` and this is where things go wrong.
  You reverted the removal of apt's index, so docker will use whatever index was stored from previous layers.
  `apt` tries to download the package from its index and.... it doesn't exist anymore because it was updated in debian's repositories.

This is NOT a nitpick comment. It effectively renders the image impossible to build after some time (unless `--no-cache` is used), so please fix it:
- if you want to keep the different calls to `apt-get` rather than keeping the single one that was there, fine for me it doesn't matter as much.
- every time you call `apt-get install` you HAVE TO do an `apt-get update` before.
- it's not strictly required to remove apt's index, but you never need it so there is no reason to keep it: `rm -rf /var/lib/apt/lists/*`

Reply:
Understood, Fixed on a9997080